components/function-controller/pkg/utils/controller_utils.go

```diff
@@ -38,79 +54,68 @@ type DefaultConfig struct {
-		return nil, err
-	}
-	var availableRuntimes []RuntimesSupported
-	if runtimeImages, ok := config.Data["runtimes"]; ok {
-		err := yaml.Unmarshal([]byte(runtimeImages), &availableRuntimes)
-		if err != nil {
-			log.Error(err, "Unable to get the supported runtimes")
-			return nil, err
-		}
-		rnInfo.AvailableRuntimes = availableRuntimes
+	mandatoryConfKeys := []string{
+		"dockerRegistry",
+		"serviceAccountName",
+		"runtimes",
+		"defaults",
+		"funcTypes",
+		"funcSizes",
 	}
 
-	if sa, ok := config.Data["serviceAccountName"]; ok {
-		rnInfo.ServiceAccount = sa
-	} else {
-		err := errors.New("Error while fetching Service Account Name")
-		log.Error(err, "Error while fetching Service Account Name")
+	if err := validateMandatoryConfig(config, mandatoryConfKeys); err != nil {
 		return nil, err
 	}
 
+	var runtimes []RuntimesSupported
+	if err := yaml.Unmarshal([]byte(config.Data["runtimes"]), &runtimes); err != nil {
+		return nil, fmt.Errorf("unable to read supported runtimes: %s", err)
+	}
+
 	var defaultConfig DefaultConfig
-	if defaults, ok := config.Data["defaults"]; ok {
-		err := yaml.Unmarshal([]byte(defaults), &defaultConfig)
-		if err != nil {
-			log.Error(err, "Error while fetching defaults")
-			return nil, err
-		}
-		rnInfo.Defaults = defaultConfig
+	if err := yaml.Unmarshal([]byte(config.Data["defaults"]), &defaultConfig); err != nil {
+		return nil, fmt.Errorf("unable to read default function config: %s", err)
 	}
 
 	var funcTypes []FuncType
-	if funcs, ok := config.Data["funcTypes"]; ok {
-		err := yaml.Unmarshal([]byte(funcs), &funcTypes)
-		if err != nil {
-			log.Error(err, "Error while fetching function types")
-			return nil, err
-		}
-		rnInfo.FuncTypes = funcTypes
+	if err := yaml.Unmarshal([]byte(config.Data["funcTypes"]), &funcTypes); err != nil {
+		return nil, fmt.Errorf("unable to read function types: %s", err)
 	}
 
 	var funcSizes []FuncSize
-	if sizes, ok := config.Data["funcSizes"]; ok {
-		err := yaml.Unmarshal([]byte(sizes), &funcSizes)
-		if err != nil {
-			log.Error(err, "Error while fetching function sizes")
-		}
-		rnInfo.FuncSizes = funcSizes
+	if err := yaml.Unmarshal([]byte(config.Data["funcSizes"]), &funcSizes); err != nil {
+		return nil, fmt.Errorf("unable to read function sizes: %s", err)
 	}
-	return rnInfo, nil
+
+	return &RuntimeInfo{
+		RegistryInfo:      config.Data["dockerRegistry"],
+		ServiceAccount:    config.Data["serviceAccountName"],
+		AvailableRuntimes: runtimes,
+		Defaults:          defaultConfig,
+		FuncTypes:         funcTypes,
+		FuncSizes:         funcSizes,
+	}, nil
 }
 
-func (ri *RuntimeInfo) DockerFileConfigMapName(runtime string) string {
-	result := ""
-	for _, runtimeInf := range ri.AvailableRuntimes {
-		if runtimeInf.ID == runtime {
-			result = runtimeInf.DockerFileName
-			break
+// validateMandatoryConfig asserts that all required values are defined in the
+// given ConfigMap.
+func validateMandatoryConfig(c *corev1.ConfigMap, mandatoryKeys []string) error {
```

Comment:
Do we have a test for this?

Reply:
Yes, not as a dedicated test but we do test with missing fields, as well as the expected error message.